src/libasr/codegen/llvm_utils.cpp

```diff
@@ -920,24 +962,20 @@ namespace LFortran {
                 llvm_utils->create_ptr_gep(src_key_mask, itr));
             LLVM::CreateStore(*builder, key_mask_value,
                 llvm_utils->create_ptr_gep(dest_key_mask, itr));
-            llvm::Function *fn = builder->GetInsertBlock()->getParent();
-            llvm::BasicBlock *thenBB = llvm::BasicBlock::Create(context, "then", fn);
-            llvm::BasicBlock *elseBB = llvm::BasicBlock::Create(context, "else");
-            llvm::BasicBlock *mergeBB = llvm::BasicBlock::Create(context, "ifcont");
+
             llvm::Value* is_key_set = builder->CreateICmpEQ(key_mask_value,
                 llvm::ConstantInt::get(llvm::Type::getInt8Ty(context), llvm::APInt(8, 1)));
-            builder->CreateCondBr(is_key_set, thenBB, elseBB);
-            builder->SetInsertPoint(thenBB);
-            {
-
-                llvm::Value* srci = llvm_utils->create_ptr_gep(src_key_value_pairs, itr);
-                llvm::Value* desti = llvm_utils->create_ptr_gep(dest_key_value_pairs, itr);
-                deepcopy_key_value_pair_linked_list(srci, desti, dest_key_value_pairs,
-                    src_capacity, dict_type, module);
-            }
-            builder->CreateBr(mergeBB);
-            llvm_utils->start_new_block(elseBB);
-            llvm_utils->start_new_block(mergeBB);
+            llvm_utils->create_if_else(
+                is_key_set,
+                [this, src_key_value_pairs, dest_key_value_pairs, itr,
+                        src_capacity, dict_type, &module]() {
```

Comment:
- The above variables inside the square bracket are read-only.
- I can also replace it with the below suggestion (but it makes all the variables read-write):
```suggestion
                [&]() {
```
Both of them work fine, but the second one reduces the pain to define the variable inside the `[]`
Which one should I use, make the variables read-only or simply use `&`?
cc @certik, @czgdp1807